respx/models.py

```diff
@@ -379,21 +354,92 @@ def match(self, request: Request) -> Optional[Union[Request, ResponseTemplate]]:
                 result = request
             return result
 
-        request_method, _request_url, *_ = request
+        request_method, request_url, *_ = request
         if self.method != request_method.decode():
             return None
 
-        if not self.url:
-            matches = True
-        elif isinstance(self.url, httpx.URL):
-            matches = self.url.raw == _request_url
-        else:
-            match = self.url.match(str(httpx.URL(_request_url)))
-            if match:
-                matches = True
-                url_params = match.groupdict()
+        matches, url_params = self.url.matches(request_url)
 
         if matches:
             return self.response.clone(context={"request": request, **url_params})
 
         return None
+
+
+class URLPattern:
+    def __init__(
+        self,
+        url: URLPatternTypes,
+        base: Optional[str] = None,
+        params: Optional[QueryParamTypes] = None,
+    ):
+        self._regex: Optional[Pattern[str]] = None
+        self._url: Optional[httpx.URL] = None
+        self._params: Optional[httpx.QueryParams] = None
+
+        url = url or None
+
+        if not base:
+            base_url = httpx.URL("")
+        else:
+            base_url = httpx.URL(base if base.endswith("/") else base + "/")
+
+            if not url:
+                url = base_url
+
+        if params is not None:
+            self._params = httpx.QueryParams(params)
+
+        if isinstance(url, (str, tuple, httpx.URL)):
+            self._url = base_url.join(httpx.URL(url))
```

Comment:
Here will `base_url` be joined with it self, when `url` arg is `None`, due to line `388`.

Maybe not a problem?

Reply:
Oh, I'll take a look